### PR TITLE
Migrate release process to github actions

### LINKED ISSUE #2621: Build snapshots from develop branch
#### Summary

We should start building snapshots from the `develop` branch.

#### Why do we need this?

To make it easier to test dev builds

#### What is already there? What do you see now?

Snapshots are only built from `master`

#### What is missing? What do you want to see?

Snapshots also being built from `develop`

#### Environment

Travis, in the future Github Actions

#### How do you propose to implement this?

Change `.travis.yml` to match against master or develop for these builds

#### How do you propose to test this?

The entire point is that this only runs on `master` and `develop`, so if we want to test this before merging, we'd have to do it in a fork or something.

### PR DESCRIPTION
**Pull request by rvolosatovs**
#### Summary

Closes #2621 
Refs #2280 

#### Changes

- Migrate release process to github actions
- Push `master` and `develop` snapshots to Docker registry

#### Testing

Fork

See action in action:
- `develop` snapshot: https://github.com/rvolosatovs/lorawan-stack/actions/runs/141058966 (actual docker push fails because of auth, which is expected. I removed documentation upload as well, since it takes enormous amount of time)
- `tag`: https://github.com/rvolosatovs/lorawan-stack/actions/runs/141073199

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: .github/workflows/release-snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    - name: Check for diff
+      run: tools/bin/mage git:diff
+    - name: Determine Goreleaser version
+      id: goreleaser_version
+      run: echo "::set-output name=value::$(cat tools/go.mod | grep 'github.com/goreleaser/goreleaser v' | cut -d ' ' -f 2)"
+    - name: Run Goreleaser
+      uses: goreleaser/goreleaser-action@v2
+      env:
+        DOCKER_IMAGE: TheThingsNetwork/lorawan-stack
+        DOCKER_IMAGE_DEV: TheThingsNetwork/lorawan-stack-dev
```
Comment (johanstokking) [inline .github/workflows/release-snapshot.yml:154]: `thethingsnetwork` (lowercase)

File: .github/workflows/release-snapshot.yml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      with:
+        version: ${{ steps.goreleaser_version.outputs.value }}
+        args: release --snapshot
+    - name: Push Docker image
+      run: |
+        docker push TheThingsNetwork/lorawan-stack-dev:${{ github.sha }}
```
Comment (johanstokking) [inline .github/workflows/release-snapshot.yml:160]: Same

File: .github/workflows/release-tag.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    - name: File versioning
+      run: tools/bin/mage version:files
+    - name: Check for diff
+      run: tools/bin/mage git:diff
+    - name: Install documentation dependencies
+      run: tools/bin/mage docs:deps
+    - name: Build documentation
+      run: tools/bin/mage docs:build
+      env:
+        HUGO_BASE_URL: https://thethingsstack.io/
```
Comment (htdvisser) [inline .github/workflows/release-tag.yml:48]: We also need `HUGO_PARAMS_SEARCH_ENABLED=true`, `HUGO_PARAMS_SEARCH_INDEX=thethingsstack` and `HUGO_PARAMS_SEARCH_APIKEY=<secret>`.

File: .github/workflows/release-tag.yml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      run: tools/bin/mage jsSDK:clean jsSDK:build
+    - name: Install JS dependencies
+      run: tools/bin/mage js:deps
+    - name: Build frontend
+      run: tools/bin/mage js:clean js:build
+    - name: File versioning
+      run: tools/bin/mage version:files
+    - name: Check for diff
+      run: tools/bin/mage git:diff
+    - name: Determine date
```
Comment (htdvisser) [inline .github/workflows/release-tag.yml:101]: We should also build the offline docs here

File: .github/workflows/release-snapshot.yml [hunk 5]
```diff
@@ -0,0 +1,160 @@
+name: Snapshot release
+
+on:
+  push:
+    branches:
+      - master
+      - develop
+    tags:
+      - 'v3.[0-9]+.[0-9]+*'
+      - '!v3.[0-9]+.[0-9]+'
```
[the remark this answers is not in the document]
Comment (rvolosatovs) [reply .github/workflows/release-snapshot.yml:10]: There are non-final release tags, e.g. rc tags

File: .github/workflows/release-snapshot.yml [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+env:
+  NODE_ENV: production
+
+jobs:
+  documentation:
```
[the remark this answers is not in the document]
Comment (rvolosatovs) [reply .github/workflows/release-snapshot.yml:16]: We make sure it builds, before it was pushed to artifacts, but it takes enormous amount of time

File: .github/workflows/release-tag.yml (hunk 4 above)
Comment (rvolosatovs) [reply .github/workflows/release-tag.yml:101]: Can you clarify? It's not something present in `.travis.yml`, right?

Comment (htdvisser) [reply .github/workflows/release-tag.yml:104]: It's `$MAGE docs:build`

File: .github/workflows/release-snapshot.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (htdvisser) [reply .github/workflows/release-snapshot.yml:10]: Okay, then perhaps good to add a `-rc[0-9]+` suffix?

File: .github/workflows/release-snapshot.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (htdvisser) [reply .github/workflows/release-snapshot.yml:16]: But we already verify documentation builds in the pull requests, right? Then why again after merge?

File: .github/workflows/release-snapshot.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (rvolosatovs) [reply .github/workflows/release-snapshot.yml:10]: Why? We want to make a snapshot build for anything that's not a proper release tag, e.g. `v3.9.9-test`, `v3.9.9-rc2` etc. but not `v3.9.9`

[the remark this answers is not in the document]
Comment (htdvisser) [reply .github/workflows/release-snapshot.yml:10]: Okay I didn't know that we wanted that.

**Pull request merged**